### DISCUSSION TITLE
Edit form with axios

### DISCUSSION THREAD

**Discussion by leandrocunha526** [react-hook-form/react-hook-form#6623] [General] [answered]
Hi,

Having a bug or is it some change that prevents you from using the value in the input to display a data in value to edit form with a set value?

Example https://codesandbox.io/s/react-hook-form-v7-getvalues-2eioh

In version 6 I remember this working.

It was not necessary to use value in input in my test in version 6.

https://react-hook-form.com/migrate-v6-to-v7/

```
import React, { useEffect } from "react";
import api from "../../services/api";
import { withRouter, useParams } from "react-router-dom";
import { Container, Form } from "./styles/EditUser";
import {useForm} from "react-hook-form";

const EditUser = (props) => {

    const {id} = useParams();

    const {register, handleSubmit, setValue} = useForm({});

    useEffect(() => {
      api.get(`/user/${id}`).then((result) => {
        setValue("id", result.data.id);
        setValue("name", result.data.name);
        setValue("username", result.data.username);
        setValue("email", result.data.email);
        setValue("password", result.data.password);
      });
    });

    const onSubmit = (data) => {
      api.put(`/user/${id}`, data).then(() => {
        props.history.push("/users");
      }).catch(error => {
        console.log(error)
      })
    }


    return (
      <main>
        <Container>
          <Form  onSubmit={handleSubmit(onSubmit)}>
            <h1>Editar usuário</h1>
            <label>Nome</label>
            <input
              type="text"
              name="name"
              {...register("name")}
            />

           <label>Nome de usuário</label>
            <input
              type="text"
              name="username"
              {...register("username")}
            />

            <label>Email</label>
            <input
              type="email"
              name="email"
              {...register("email")}
            />

            <label>Senha</label>
            <input
              type="password"
              name="password"
              {...register("password")}
            />

            <button type="submit">Enviar</button>
          </Form>
        </Container>
      </main>
    );
}

export default withRouter(EditUser);
```

**Comment by leandrocunha526** [comment]
Take me out of this doubt please @bluebill1049. Thanks!

**Reply by bluebill1049** [reply]
> It was not necessary to use value in input in my test in version 6. 

what do you mean by that?

**Reply by leandrocunha526** [reply]
@bluebill1049 Would there be any way in version 7 to assign values to form inputs? In version 6 I didn't need anything other than what I inserted in this code.

**Reply by bluebill1049** [reply]
you can use `defaultValues` right? is that what you try to achieve?

**Reply by leandrocunha526** [reply]
@bluebill1049 I need it to be already populated with values in the input when getting a given userid. I've already done this on Vue and on the previous version of the framework which helped me a lot in the implementation.

**Comment by slugmandrew** [comment, marked as answer]
Not sure if it's related but did you check when your `useEffect` is running? It might be trying to call `setValue` before the form is ready. Perhaps try putting that in the dependency array, or do you want to run this on every render?

**Reply by bluebill1049** [reply]
good point and why is `useEffect` without deps as well?

**Comment by leandrocunha526** [comment]
Fixed it. Thanks for help.
